File: src/main/java/ovh/paulem/btm/config/ConfigBlacklist.java
package ovh.paulem.btm.config;

import org.bukkit.Material;
import org.bukkit.entity.Player;
import ovh.paulem.btm.BetterMending;

import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

public class ConfigBlacklist {
    private final List<String> blacklistedPlayers;
    private final List<Material> blacklistedItems;

    public ConfigBlacklist() {
        blacklistedPlayers = BetterMending.getConf().getStringList("blacklisted-players")
                .stream()
                .map(String::toLowerCase)
                .collect(Collectors.toList());

        blacklistedItems = BetterMending.getConf().getStringList("blacklisted-items")
                .stream()
                .map(s -> Material.getMaterial(s.toUpperCase()))
                .filter(Objects::nonNull)
                .collect(Collectors.toList());
    }

    public boolean isBlacklisted(Player player) {
        return blacklistedPlayers.contains(player.getName().toLowerCase());
    }

    // TODO : Add support for Oraxen items
    public boolean isBlacklisted(Material material) {
        return blacklistedItems.contains(material);
    }
}

File: src/main/kotlin/net/paulem/btm/config/ConfigBlacklist.kt
package net.paulem.btm.config

import net.paulem.btm.BetterMending
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

class ConfigBlacklist {
    val blacklistedPlayers: List<String>
    val blacklistedItems: List<String>
    val blacklistedLores: List<String>

    init {
        val config = BetterMending.instance.config

        blacklistedPlayers = config.getStringList("blacklisted-players")
            .map { it.lowercase() }

        blacklistedItems = config.getStringList("blacklisted-items")

        blacklistedLores = config.getStringList("blacklisted-lores")
            .map { it.lowercase() }
    }

    fun isBlacklisted(player: Player): Boolean {
        return blacklistedPlayers.contains(player.name.lowercase())
    }

    fun isBlacklisted(stack: ItemStack): Boolean {
        return BetterMending.oraxenCompat.isBlacklisted(stack)
    }
}